segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainer.java

```diff
@@ -58,4 +62,20 @@
         ArrayView segmentInfo = MetadataStore.SegmentInfo.recoveredSegment(streamSegmentName, length, isSealed);
         return metadataStore.createSegment(streamSegmentName, segmentInfo, new TimeoutTimer(TIMEOUT));
     }
+
+    @Override
+    public void copySegment(Storage storage, String sourceSegment, String targetSegment)
+            throws Exception {
+        storage.create(targetSegment, TIMEOUT);
```

Comment:
This is an async method. And so is `getStreamSegmentInfo`.

Please rewrite this entire method to be aynchronous:
1. `public CompletableFuture<Void> copySegment(...)` (remove the `throws Exception`).
2. Use `CompletableFuture` composition (`thenCompose`, `thenApply`, etc.) to chain your futures together. 
3. Use `Futures.loop`
4. return the resulting `CompletableFuture`.

Reply:
Done.